subsys/secure_storage/Kconfig.its_transform

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2024
+# SPDX-License-Identifier: Apache-2.0
+
+choice SECURE_STORAGE_ITS_TRANSFORM_IMPLEMENTATION
+	prompt "ITS transform module implementation"
+
+config SECURE_STORAGE_ITS_TRANSFORM_IMPLEMENTATION_AEAD
+	bool "ITS transform module implementation using AEAD to protect the data"
+
+config SECURE_STORAGE_ITS_TRANSFORM_IMPLEMENTATION_CUSTOM
+	bool "Custom ITS transform module implementation"
+	help
+	  Implement the functions declared in <zephyr/secure_storage/its/transform.h>
+	  and set CONFIG_SECURE_STORAGE_ITS_TRANSFORM_OUTPUT_OVERHEAD appropriately.
+
+endchoice # SECURE_STORAGE_ITS_TRANSFORM_IMPLEMENTATION
+
+config SECURE_STORAGE_ITS_TRANSFORM_OUTPUT_OVERHEAD
+	int "Overhead, in bytes, associated with the transformation of an entry's data for storage"
+	range 0 1000
+	# nonce (12) + authentication tag (16)
+	default 28 if SECURE_STORAGE_ITS_TRANSFORM_IMPLEMENTATION_AEAD
+	default -1
+	help
+	  This indicates how many more bytes an ITS entry's data will be once it
+	  has been processed by the secure_storage_its_transform_to_store() function.
+
+if SECURE_STORAGE_ITS_TRANSFORM_IMPLEMENTATION_AEAD
+
+choice SECURE_STORAGE_ITS_TRANSFORM_AEAD_SCHEME_PROVIDER
+	prompt "AEAD ITS transform module scheme"
+	help
+	  The AEAD scheme used to encrypt and authenticate the data.
+
+config SECURE_STORAGE_ITS_TRANSFORM_AEAD_SCHEME_PROVIDER_DEFAULT
+	prompt "Default AEAD scheme (AES-GCM)"
+	select PSA_WANT_KEY_TYPE_AES
+	select PSA_WANT_ALG_GCM
+
+config SECURE_STORAGE_ITS_TRANSFORM_AEAD_SCHEME_PROVIDER_CUSTOM
+	bool "Custom AEAD scheme"
+	help
+	  Implement the secure_storage_its_transform_aead_get_scheme() function
+	  and set CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_NONCE_SIZE appropriately.
+
+endchoice # SECURE_STORAGE_ITS_TRANSFORM_AEAD_SCHEME_PROVIDER
+
+choice SECURE_STORAGE_ITS_TRANSFORM_AEAD_KEY_PROVIDER
+	prompt "AEAD ITS transform module encryption key provider"
+
+config SECURE_STORAGE_ITS_TRANSFORM_AEAD_KEY_PROVIDER_DEFAULT
```

Comment:
Assuming, we might add additional implementations in the future. What do you think about renaming it more descriptively to something like SECURE_STORAGE_ITS_TRANSFORM_AEAD_KEY_PROVIDER_HWINFO